Flagsmith.FlagsmithClient/Cache/IdentityFlagListCache.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Flagsmith.Providers;
+
+namespace Flagsmith.Cache
+{
+    internal class IdentityFlagListCache : FlagListCache
+    {
+        private readonly IdentityTraitsKey _identityTraitsKey;
+
+        public IdentityFlagListCache(IdentityTraitsKey identityTraitsKey, IFlags flags, IDateTimeProvider dateTimeProvider, int cacheDurationInMinutes) :
+            base(dateTimeProvider, flags, cacheDurationInMinutes)
+        {
+            _identityTraitsKey = identityTraitsKey;
+        }
+
+        public IFlags GetLatestFlags(GetIdentityFlagsDelegate getFlagsDelegate)
```

Comment:
Perhaps this is my limited knowledge of C# but I'm a bit confused here as to why we're passing in the GetIdentityFlagsDelegate object that seems to already be defined on the abstract class? This seems like it could cause unexpected side effects. Could you explain the logic here and what this Delegate object's intended behaviour is?

Reply:
You can see my comment here where I explain what is the goal of a delegate: https://github.com/Flagsmith/flagsmith-dotnet-client/pull/77/files#r1363795518

Let me know if you need more information